libs/chain-signatures/contract/src/lib.rs

```diff
@@ -62,19 +62,76 @@ impl Default for VersionedMpcContract {
     }
 }
 
+pub trait SignatureStateHandler {
```

Comment:
It would be more elegant if `remove_request` were part of this trait alongside add and get.